Test report CI job failing

We've been getting 

```python
Traceback (most recent call last):
Getting list of workflow runs...
Found [15](https://github.com/dask/distributed/actions/runs/4424054212/jobs/7757420843#step:7:16)1 workflow runs
  File "/home/runner/work/distributed/distributed/continuous_integration/scripts/test_report.py", line 513, in <module>
Fetching artifact listing for the 30 most recent workflow runs
    main()
Downloading and parsing 805 artifacts...
  File "/home/runner/work/distributed/distributed/continuous_integration/scripts/test_report.py", line 397, in main
    dfs = list(
  File "/home/runner/work/distributed/distributed/continuous_integration/scripts/test_report.py", line 367, in download_and_parse_artifacts
    assert (
AssertionError: Artifact suite name ubuntu-latest-mindeps--ci1 did not match any jobs dataframe:
[nan 'ubuntu-latest-mindeps-queue-ci1'
 'ubuntu-latest-mindeps-queue-notci1' 'ubuntu-latest-3.8-queue-ci1'
 'ubuntu-latest-3.8-queue-notci1' 'ubuntu-latest-3.9-queue-ci1'
 'ubuntu-latest-3.9-queue-notci1' 'ubuntu-latest-3.10-queue-ci1'
 'ubuntu-latest-3.10-queue-notci1' 'ubuntu-latest-3.11-queue-ci1'
 'ubuntu-latest-3.11-queue-notci1' 'windows-latest-3.8-queue-ci1'
 'windows-latest-3.8-queue-notci1' 'windows-latest-3.9-queue-ci1'
 'windows-latest-3.9-queue-notci1' 'windows-latest-3.10-queue-ci1'
 'windows-latest-3.10-queue-notci1' 'windows-latest-3.11-queue-ci1'
 'windows-latest-3.11-queue-notci1' 'macos-latest-3.8-queue-ci1'
 'macos-latest-3.8-queue-notci1' 'macos-latest-3.11-queue-ci1'
 'macos-latest-3.11-queue-notci1' 'ubuntu-latest-3.9-no_queue-ci1'
 'ubuntu-latest-3.9-no_queue-notci1' 'ubuntu-latest-mindeps-numpy-ci1'
 'ubuntu-latest-mindeps-numpy-notci1']
```

in our test report jobs for a while (see [this build](https://github.com/dask/distributed/actions/runs/4424054212/jobs/7757420843) for an example)

--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -71,9 +71,9 @@ jobs:
       - name: Set $TEST_ID
         run: |
           export PARTITION_LABEL=$( echo "${{ matrix.partition }}" | sed "s/ //" )
-          export TEST_ID="${{ matrix.os }}-${{ matrix.environment }}-${{ matrix.queuing }}-$PARTITION_LABEL"
+          export TEST_ID="${{ matrix.os }}-${{ matrix.environment }}-${{ matrix.queuing || join(matrix.extra_packages, '_') }}-$PARTITION_LABEL"
           # Switch to this version for stress-test:
-          # export TEST_ID="${{ matrix.os }}-${{ matrix.environment }}-${{ matrix.queuing }}-$PARTITION_LABEL-${{ matrix.run }}"
+          # export TEST_ID="$TEST_ID-${{ matrix.run }}"
           echo "TEST_ID: $TEST_ID"
           echo "TEST_ID=$TEST_ID" >> $GITHUB_ENV
         shell: bash

--- a/continuous_integration/scripts/test_report.py
+++ b/continuous_integration/scripts/test_report.py
@@ -215,10 +215,6 @@ def suite_from_name(name: str) -> str:
     just lop off the front of the name to get the suite.
     """
     parts = name.split("-")
-    if len(parts) == 4:  # [OS, 'latest', py_version, $PARTITION_LABEL]
-        # Migration: handle older jobs without the `queuing` configuration.
-        # This branch can be removed after 2022-12-07.
-        parts.insert(3, "no_queue")
     return "-".join(parts[:4])
 
 

--- a/continuous_integration/scripts/test_report.py
+++ b/continuous_integration/scripts/test_report.py
@@ -359,6 +359,10 @@ def download_and_parse_artifacts(
                 if xml is None:
                     continue
                 df = dataframe_from_jxml(cast(Iterable, xml))
+
+                # Needed until *-*-mindeps-numpy shows up in TEST_ID
+                a["name"] = a["name"].replace("--", "-numpy-")
+
                 # Note: we assign a column with the workflow run timestamp rather
                 # than the artifact timestamp so that artifacts triggered under
                 # the same workflow run can be aligned according to the same trigger

